### PR TITLE
feat: buffered events file reader

### LINKED ISSUE #7517: Core File Client - Roll files
The cores file client currently persists all events to a single file.  This means there is no way to compress and cleanup storage as the file cannot be cleanly split (without a process to read it in and do this).  If we make the file client split the file after X events (such that we get reasonable event file size say ~ 100MB uncompressed which should be about 10MB compressed), then it will easy to add a job to cleanup and compress these as we go meaning that, if we cannot always have all event data as it may be too large to store from block 0, we can at least have the most recent few days data which will make it easier to diagnose issues.   Related to this, the data-node file reader would need to change to sequentially read in a time sorted collection of event files.

### PR DESCRIPTION
**Pull request by ettec**
closes #7517 adds the reader for event buffer files and removes the legacy event file reader.  Consolidates all event files to conform with the buffer file format so we only have one format.

### CODE REVIEW

File: core/broker/file_client.go [hunk 1]
```diff
@@ -73,19 +78,38 @@ func (fc *FileClient) Send(event events.Event) error {
...
-	sizeBytes := make([]byte, 4)
-	binary.BigEndian.PutUint32(sizeBytes, size)
+	if err != nil {
+		return fmt.Errorf("failed to write event to buffer file: %w", err)
+	}
+
+	return nil
+}
+
+func WriteToBufferFile(bufferFile *os.File, bufferSeqNum uint64, event events.Event) error {
```
[the remark this answers is not in the document]
Comment (ettec) [reply core/broker/file_client.go:91]: the file is used only for testing, this ensure we have one format across datanode and core

**Review by EVODelavega — DISMISSED** [review]
The core shouldn't produce event files. If we need them, we should dump the events the core emits by subscribing to the event stream/queue and persisting them there

File: core/broker/file_client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (EVODelavega) [reply core/broker/file_client.go:91]: All the more reason the code shouldn't sit in the core

**Review by pscott31 — COMMENTED** [review]
Looks alright to me, though I'm curious as to what Elias' objection is about.

I think if it were me rather than having to specify a directory that only contains event files, I might have made it so it looks for some sort of next-numbered file or something e.g. `vega.evt.001` -> `vega.evt.002` but I'm not actually sure that is that much nicer and it's a rather minor thing anyway

**Comment by ettec** [comment]
> The core shouldn't produce event files. If we need them, we should dump the events the core emits by subscribing to the event stream/queue and persisting them there



> The core shouldn't produce event files. If we need them, we should dump the events the core emits by subscribing to the event stream/queue and persisting them there

Summary of discussion just now->  view on balance on discussion with Phil is its better to keep this functionality in place in core in  case we encounter an issue such as has been encounted in the past where the event stream received by datanode was different to that being produced by core.  So the change here is to existing code  just to make the format consistent for event files, such that we only need one reader and don't have compatibility issues.

**Comment by pscott31** [comment]
> The core shouldn't produce event files. If we need them, we should dump the events the core emits by subscribing to the event stream/queue and persisting them there

But it already does, and it's massively helpful for debugging and developing (e.g. when the system tests run, they generate an event file that I can play directly through datanode to recreate what happened)

**Review by pscott31 — DISMISSED** [review]

**Review by pscott31 — APPROVED** [review]

**Pull request merged**